File: frontend/src/services/CustomerService.js
import axiosInstance from './axiosConfig';

const CustomerService = {
  // Get all customers with pagination
  getAllCustomers: (page = 0, size = 10, sortBy = 'id', sortDir = 'ASC') => {
    return axiosInstance.get('/customers', {
      params: { page, size, sortBy, sortDir }
    });
  },

  // Search customers
  searchCustomers: (keyword, page = 0, size = 10) => {
    return axiosInstance.get('/customers/search', {
      params: { keyword, page, size }
    });
  },

  // Get customer by ID
  getCustomerById: (id) => {
    return axiosInstance.get(`/customers/${id}`);
  },

  // Create new customer
  createCustomer: (customerData) => {
    return axiosInstance.post('/customers', customerData);
  },

  // Update customer
  updateCustomer: (id, customerData) => {
    return axiosInstance.put(`/customers/${id}`, customerData);
  },

  // Delete customer
  deleteCustomer: (id) => {
    return axiosInstance.delete(`/customers/${id}`);
  },

  // Import customers from Excel
  importCustomers: (file) => {
    const formData = new FormData();
    formData.append('file', file);
    
    return axiosInstance.post('/customers/import', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    });
  },

  // Export customers to Excel
  exportCustomers: () => {
    return axiosInstance.get('/customers/export', {
      responseType: 'blob'
    });
  },

  // Add family member
  addFamilyMember: (customerId, familyMemberId) => {
    return axiosInstance.post(`/customers/${customerId}/family-members/${familyMemberId}`);
  },

  // Remove family member
  removeFamilyMember: (customerId, familyMemberId) => {
    return axiosInstance.delete(`/customers/${customerId}/family-members/${familyMemberId}`);
  }
};

export default CustomerService;

File: backend/src/main/java/com/cms/customer/service/CustomerService.java
package com.cms.customer.service;

import com.cms.customer.dto.CustomerDTO;
import com.cms.customer.dto.ImportResultDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface CustomerService {

    CustomerDTO createCustomer(CustomerDTO customerDTO);

    CustomerDTO updateCustomer(Long id, CustomerDTO customerDTO);

    CustomerDTO getCustomerById(Long id);

    Page<CustomerDTO> getAllCustomers(Pageable pageable);

    Page<CustomerDTO> searchCustomers(String keyword, Pageable pageable);

    void deleteCustomer(Long id);

    ImportResultDTO importCustomersFromExcel(MultipartFile file);

    byte[] exportCustomersToExcel();

    void addFamilyMember(Long customerId, Long familyMemberId);

    void removeFamilyMember(Long customerId, Long familyMemberId);
}
